qutebrowser/config/config.py

```diff
@@ -379,7 +386,7 @@ def get_obj(self,
         Note that the returned values are not watched for mutation.
         If a URL is given, return the value which should be used for that URL.
         """
-        self.get_opt(name)  # To make sure it exists
+        assert self.has_opt(name)
```

Comment:
This breaks things because it doesn't raise `NoOptionError` anymore, and also assertions can be disabled with `python -O`.

Why not add a `self.ensure_has_opt(name)` (could even be private?) which also sets `deleted`/`renamed`? Basically, move the `except KeyError:` part of `get_opt` into its own function, then that can also be used in `get_opt` itself.

Reply:
I commented about the change break some part of the code at https://github.com/qutebrowser/qutebrowser/pull/4808#issuecomment-497635407 .

About `ensure_has_opt`: okay. It keeps the current interface (raise `NoOptionError` when there's no option, the caller can catch that)

Reply:
Ah, I see that's basically what you proposed in https://github.com/qutebrowser/qutebrowser/pull/4808#issuecomment-497615572 - after looking at things in detail, I agree a `ensure_has_opt` would be the right thing to do.

Note that there's more than one (non-test) place handling `NoOptionError`: It inherits `configexc.Error`, and that is handled in a lot of places.